src/io/output_vtk.cc

```diff
@@ -186,7 +183,8 @@ void OutputVTK::write_vtk_vtu_head(void)
 
 
 
-void OutputVTK::fill_element_types_vector(std::vector< unsigned int >& data)
+template <class T>
+void OutputVTK::fill_element_types_vector(std::vector< T >& data)
```

Comment:
Use uint8_t instead of unsigned int here, remove template. Fix MeshData::print_ascii to check sizeof(T) ==1 and in such case cast to int or unsigned int according to type trait is_signed<T>::value